### DISCUSSION TITLE
Bug found in ADF ingestion pipeline when you have duplicate filenames

### DISCUSSION THREAD

**Discussion by thecloudman** [microsoft/finops-toolkit#619] [Q&A]
Been doing testing around why my billed cost was not matching exports from usage details. I found that when doing onetime exports, if you had duplicate filenames in your exports, for example you split your July onetime export, you would have part_0_000_01.csv in one folder and part_0_000_01.csv in another folder. When you run the ingestion pipeline, the convert to csv will overwrite data in the existing filename. To overcome this i changed the Set FileName variable from: **@replace(pipeline().parameters.fileName, '.csv', '.parquet'**) to **@concat(replace(pipeline().parameters.fileName, '.csv', ''), '_', formatDateTime(utcNow(), 'yyyyMMddHHmmss'), '.parquet')**. This appends a date format to the end of the file and creates a unique filename for each imported file. It prevents overwriting the original parquet file.

**Comment by flanakin** [comment]
What version is this for? We found a bug in 0.1 where one-time exports would go to a different folder and fixed that in 0.2.

**Comment by thecloudman** [comment]
no longer an issue
